src/Rbac.php

```diff
@@ -139,20 +139,22 @@ public function getRole($objectOrName)
      */
     public function isGranted($role, $permission, $assert = null)
     {
+        $result = $this->getRole($role)->hasPermission($permission);
+
         if ($assert) {
             if ($assert instanceof AssertionInterface) {
-                return (bool) $assert->assert($this);
+                return $result && $assert->assert($this);
```

Comment:
Then you don't have to test $result here anymore. just return the result of the assertion check